src/main/java/org/mondemand/Client.java

```diff
@@ -316,6 +315,14 @@ public void setErrorHandler(ErrorHandler errorHandler) {
     }
   }
 
+  /**
+   * @param maxNumMetrics the maximum number of metrics to send in stats messages
+   */
+  public void setMaxNumMetrics(Integer maxNumMetrics)
+  {
+    this.maxNumMetrics = maxNumMetrics;
```

Comment:
check for <=0 values.
also, we may want to define a MAX value for this and check >MAX_.. too.